components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1118,6 +1120,99 @@ public OAuthRevocationResponseDTO revokeAuthzForAppsByResourceOwner(
         return new OAuthRevocationResponseDTO();
     }
 
+    /**
+     * Revoke issued tokens for the application.
+     *
+     * @param application {@link OAuthAppRevocationRequestDTO}
+     * @return revokeRespDTO {@link OAuthAppRevocationRequestDTO}
+     * @throws IdentityOAuthAdminException Error while revoking the issued tokens.
+     */
+    public OAuthRevocationResponseDTO revokeIssuedTokensByApplication(OAuthAppRevocationRequestDTO application)
+            throws IdentityOAuthAdminException {
+
+        triggerPreApplicationTokenRevokeListeners(application);
+        String consumerKey = application.getConsumerKey();
+        String applicationName = application.getApplicationName();
+        String tenantDomain = application.getTenantDomain();
+        if (StringUtils.isNotBlank(consumerKey) && StringUtils.isNotBlank(applicationName)
+                && StringUtils.isNotBlank(tenantDomain)) {
+            List<AccessTokenDO> activeDetailedTokens;
+            try {
+                //Get active access tokens by application id
+                activeDetailedTokens = new ArrayList<>(OAuthTokenPersistenceFactory
+                        .getInstance().getAccessTokenDAO().getActiveAcessTokenDataByConsumerKey(consumerKey));
+            } catch (IdentityOAuth2Exception e) {
+                String errorMsg = "Error occurred while retrieving access tokens issued for " +
+                        "consumer key: " + consumerKey;
+                throw handleError(errorMsg, e);
+            }
+            if (activeDetailedTokens.size() > 0) {
+                String[] accessTokens = new String[activeDetailedTokens.size()];
+                List<String> tenantDomains = new ArrayList<>();
+                tenantDomains.add(tenantDomain);
+                int countToken = 0;
+                AuthenticatedUser authzUser;
+                for (AccessTokenDO detailToken : activeDetailedTokens) {
+                    String token = detailToken.getAccessToken();
+                    accessTokens[countToken++] = token;
+                    OAuthCacheKey cacheKeyToken = new OAuthCacheKey(token);
+                    String scope = buildScopeString(detailToken.getScope());
+
+                    //Clear cache with AccessTokenDO
```

Comment:
why is this comment at the place where the authorized user is retrieved?

Reply:
Addressed in [6918b82](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/commits/6918b828f239ba766e20f66e6d0af997d891413a)